packages/playwright/src/matchers/toMatchSnapshot.ts

```diff
@@ -116,15 +142,24 @@ class SnapshotHelper<T extends ImageComparatorOptions> {
       }
     }
 
-    options = {
-      ...configOptions,
-      ...options,
+    const filteredConfigOptions = configOptions;
+    for (const prop of NonConfigProperties)
+      delete (filteredConfigOptions as any)[prop];
```

Comment:
Is this deleting on the user's config options object? Perhaps make a clone?

Reply:
Sure, but I am deleting the properties that we don't want to see there, so maybe it is Ok.